Feature blog for ReadWriteOncePod GA graduation (#43418)
kubernetes/website: content/en/blog/_posts/2023-10-10-read-write-once-pod-access-mode-ga.md

Reviewer: Throughout the article, it should be `ReadWriteOncePod` (ie, inside backticks).

in the _Introducing Single Pod Access Mode for PersistentVolumes_ article from 2021.

## How can I start using ReadWriteOncePod?

ReadWriteOncePod is enabled by default in cluster versions 1.27 and beyond.

Note that ReadWriteOncePod is
[only supported for CSI volumes](/docs/concepts/storage/persistent-volumes/#access-modes).
Before using this feature you will need to update the following
[CSI sidecars](https://kubernetes-csi.github.io/docs/sidecar-containers.html)
to these versions or greater:

- [csi-provisioner:v3.0.0+](https://github.com/kubernetes-csi/external-provisioner/releases/tag/v3.0.0)
- [csi-attacher:v3.3.0+](https://github.com/kubernetes-csi/external-attacher/releases/tag/v3.3.0)
- [csi-resizer:v1.3.0+](https://github.com/kubernetes-csi/external-resizer/releases/tag/v1.3.0)

To start using ReadWriteOncePod, create a PVC with the ReadWriteOncePod access mode:

```yaml
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: single-writer-only
spec:
  accessModes:
  - ReadWriteOncePod # Allow only a single pod to access single-writer-only.
  resources:
    requests:
      storage: 1Gi
```

If your storage plugin supports

> [dynamic provisioning](/docs/concepts/storage/dynamic-provisioning/),

Reviewer 2: ```suggestion
[Dynamic provisioning](/docs/concepts/storage/dynamic-provisioning/), then 
```

> new PersistentVolumes will be created with the ReadWriteOncePod access mode applied.

Author: I find the original wording a little clearer. What about this?
> new PersistentVolumes will be created with the ReadWriteOncePod access mode set.

> new PersistentVolumes will be created with the ReadWriteOncePod access mode applied.

Reviewer: I agree: the wording originally proposed is what I prefer.

Read [Migrating existing PersistentVolumes](/blog/2021/09/13/read-write-once-pod-access-mode-alpha/#migrating-existing-persistentvolumes)
for details on migrating existing volumes to use ReadWriteOncePod.

## How can I learn more?

> Please see the [alpha](/blog/2021/09/13/read-write-once-pod-access-mode-alpha)

Reviewer 2: ```suggestion
Please see the blog posts [alpha](/blog/2021/09/13/read-write-once-pod-access-mode-alpha), 
```

> and [beta](/blog/2023/04/20/read-write-once-pod-access-mode-beta) blog posts and

Reviewer 2: ```suggestion
[beta](/blog/2023/04/20/read-write-once-pod-access-mode-beta) and 
```

---
layout: blog

> title: "Kubernetes 1.29: Single Pod Access Mode for PersistentVolumes Graduates to GA/Stable"

Reviewer: ```suggestion
title: "Kubernetes 1.29: Single Pod Access Mode for PersistentVolumes Graduates to Stable"
```

date: 2023-10-10
slug: read-write-once-pod-access-mode-ga
---

**Author:** Chris Henzie (Google)

> With the release of Kubernetes 1.29 the ReadWriteOncePod feature has graduated

Reviewer: ```suggestion
With the release of Kubernetes 1.29 the `ReadWriteOncePod` volume access mode has graduated
```

> to GA. In this blog post, we'll take a closer look at this feature and what it

Reviewer: (nit)
```suggestion
to general availability: it's part of Kubernetes' stable API.
In this blog post, I'll take a closer look at this access mode and what it
```
(just one author, so first person singular please)

does.

## What is ReadWriteOncePod?

> ReadWriteOncePod is a new access mode for

Reviewer: ```suggestion
ReadWriteOncePod is an access mode for
```
It can be new, or it can have been beta since v1.27; I don't think it can be both.

[PersistentVolumes](/docs/concepts/storage/persistent-volumes/#persistent-volumes) (PVs)
and [PersistentVolumeClaims](/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims) (PVCs)
introduced in Kubernetes 1.22. This access mode enables you to restrict volume
access to a single pod in the cluster, ensuring that only one pod can write to
the volume at a time. This can be particularly useful for stateful workloads
that require single-writer access to storage.

For more context on access modes and how ReadWriteOncePod works read
[What are access modes and why are they important?](/blog/2021/09/13/read-write-once-pod-access-mode-alpha/#what-are-access-modes-and-why-are-they-important)
in the _Introducing Single Pod Access Mode for PersistentVolumes_ article from 2021.

## How can I start using ReadWriteOncePod?

> ReadWriteOncePod is enabled by default in cluster versions 1.27 and beyond.

Reviewer: ```suggestion
The `ReadWriteOncePod` volume access mode is available by default in Kubernetes versions 1.27 and beyond.
In Kubernetes v1.29 and later, the Kubernetes API always recognizes this access mode.
```
